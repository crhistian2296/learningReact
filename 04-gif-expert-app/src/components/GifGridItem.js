import PropTypes from "prop-types";
import React from "react";

const GifGridItem = ({ title, url }) => {
    // console.log(id, title, url);
    return (
        <figure className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <figcaption className="card-img_title">{title}</figcaption>
        </figure>
    );
};

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};

export default GifGridItem;
