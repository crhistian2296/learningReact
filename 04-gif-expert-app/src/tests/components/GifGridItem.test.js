import { shallow } from "enzyme";
import React from "react";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en GifgridItem", () => {
    test("Debe mostrar el componente correctamenete", () => {
        const wrapper = shallow(<GifGridItem url="url" title="title" />);
        expect(wrapper).toMatchSnapshot();
    });
});
