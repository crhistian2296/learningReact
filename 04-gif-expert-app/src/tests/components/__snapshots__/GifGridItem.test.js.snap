// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifgridItem Debe mostrar el componente correctamenete 1`] = `
<figure
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="url"
  />
  <figcaption
    className="card-img_title"
  >
    title
  </figcaption>
</figure>
`;
